src/opm/output/eclipse/Summary.cpp

```diff
@@ -796,6 +796,47 @@ inline quantity crate( const fn_args& args ) {
     return { v, rate_unit< phase >() };
 }
 
+template <bool injection = true>
+quantity crate_resv( const fn_args& args ) {
+    const quantity zero = { 0.0, rate_unit<rt::reservoir_oil>() };
+    if (args.schedule_wells.empty())
+        return zero;
+
+    const auto& name = args.schedule_wells.front().name();
+    auto xwPos = args.wells.find(name);
+    if ((xwPos == args.wells.end()) ||
+        (xwPos->second.dynamicStatus == Opm::Well::Status::SHUT) ||
+        (xwPos->second.current_control.isProducer == injection))
```

Comment:
Maybe `...isProducer == false` would read easier?

Reply:
> Maybe `...isProducer == false` would read easier?

Perhaps, but that's the wrong criterion.  We want to return `zero` if

1. The well is a producer (`isProducer = true`), but we're asking for injection rates (`CVIR`, `injection = true`)
2. The well is an injector (`isProducer = false`), but we're asking for production rates (`CVPR`, `injection = false`)

If we wanted to spell this out with logical operators it would be something along the lines of
```C++
(isProducer && injection) || (!isProducer && !injection)
```
at which point I concluded that the brevity of `isProducer == injection` outweighed the cost of thinking the cases.  I suppose we could add a comment though.